video/out/vo_dmabuf_wayland.c

```diff
@@ -191,6 +223,15 @@ static void draw_frame(struct vo *vo, struct vo_frame *frame)
             p->want_reset = false;
     }
 
+    // 2. adjust viewport source if needed
+    struct mp_rect src;
+    struct mp_rect dst;
+    struct mp_osd_res osd;
+    mp_get_src_dst_rects(NULL, vo->opts, vo->driver->caps, vo->params,
+                         vo->dwidth, vo->dheight, vo->monitor_par,
+                         &src, &dst, &osd);
+    set_viewport_source(vo, src);
```

Comment:
Sorry, I know I just approved this, but I just realized that you can actually just move this to resize right after you're doing setting the subsurface position. No need to run this every frame.

Reply:
Thanks, but we do need to run it each frame - there can be a frame transition without a resize.

Reply:
What do you mean? If the panscan/pan change, `VOCTRL_SET_PANSCAN` will handle it.